spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/AbstractTraceHttpRequestInterceptor.java

```diff
@@ -34,38 +34,16 @@
 abstract class AbstractTraceHttpRequestInterceptor {
 
 	protected final Tracer tracer;
+	protected final SpanInjector<HttpRequest> spanInjector;
 
-	protected AbstractTraceHttpRequestInterceptor(Tracer tracer) {
+	protected AbstractTraceHttpRequestInterceptor(Tracer tracer,
+			SpanInjector<HttpRequest> spanInjector) {
 		this.tracer = tracer;
+		this.spanInjector = spanInjector;
 	}
 
-
-	private void enrichWithTraceHeaders(HttpRequest request, Span span) {
-		setIdHeader(request, Span.TRACE_ID_NAME, span.getTraceId());
-		setIdHeader(request, Span.SPAN_ID_NAME, span.getSpanId());
-		if (!span.isExportable()) {
-			setHeader(request, Span.NOT_SAMPLED_NAME, "true");
-		}
-		setHeader(request, Span.SPAN_NAME_NAME, span.getName());
-		setIdHeader(request, Span.PARENT_ID_NAME, getParentId(span));
-		setHeader(request, Span.PROCESS_ID_NAME, span.getProcessId());
-	}
-
-	private Long getParentId(Span span) {
-		return !span.getParents().isEmpty() ? span.getParents().get(0) : null;
-	}
-
-	private void setHeader(HttpRequest request, String name, String value) {
-		if (StringUtils.hasText(value) && !request.getHeaders().containsKey(name) &&
-				this.tracer.isTracing()) {
-			request.getHeaders().add(name, value);
-		}
-	}
-
-	private void setIdHeader(HttpRequest request, String name, Long value) {
-		if (value != null) {
-			setHeader(request, name, Span.idToHex(value));
-		}
+	private void enrichWithTraceHeaders(Span span, HttpRequest request) {
```

Comment:
loose the dispatch and inline. that way, we call the same operation the same thing (injecting a span into a request as opposed to enriching with trace headers)